### ISSUE TITLE
allow to add padding to tab body 

### ISSUE THREAD

**Issue by boonkerz** [Adanos020/egui_dock#1]
it should be possible to set an padding for the tab body.

before:
<img width="413" alt="image" src="https://user-images.githubusercontent.com/277321/183849829-1abf66c5-a7bb-476b-b632-e53ede7b558c.png">

after adding padding:
<img width="381" alt="image" src="https://user-images.githubusercontent.com/277321/183850496-641dca32-edd2-4c4e-b058-147cd03a2ab4.png">

currently i have hard coded it into the lib.

**Comment by Jerrody** [comment]
@boonkerz 
Can you share your lib so it is easy to add it?

**Comment by Adanos020** [comment]
I'm assuming this is meant to allow to add padding to individual tabs rather than have it as part of the `Style` struct (affecting all tabs)?

**Comment by boonkerz** [comment]
i use the same paddings in all tabs :)

**Comment by Adanos020** [comment]
Ok, then changing the value of `Style::padding` should do the trick for you.

I think being able to specify a padding on a specific tab can also be useful, so I'll keep this issue open for the time being.

**Comment by boonkerz** [comment]
> Ok, then changing the value of `Style::padding` should do the trick for you.
> 
> I think being able to specify a padding on a specific tab can also be useful, so I'll keep this issue open for the time being.

no because this affects the complete tab widget and not the tab content itself
<img width="509" alt="image" src="https://user-images.githubusercontent.com/277321/184435707-38a45a70-80b8-4dca-b53e-d8c66f1f1c04.png">

my red marks does it show :)
<img width="509" alt="image" src="https://user-images.githubusercontent.com/277321/184435925-a8b2305f-c977-4dae-a813-e2236368cf36.png">

**Comment by Adanos020** [comment]
Ah, got it now! Thanks for clarifying

**Comment by boonkerz** [comment]
yea i think the var name should be renamed to widget_margin or any thing :)
and we need content_padding or tab_content_margin should be possible

**Comment by Adanos020** [comment]
Yup, this sounds like a good idea. I can start working on this very soon, unless you want to give it a try :)

**Comment by Jerrody** [comment]
@Adanos020 
In the example exist this implementation of the `Tab`:
```rs
 impl Tab<MyContext> for PlaceholderTab {
    fn title(&self) -> &str {
        &self.title
    }

    fn ui(&mut self, ui: &mut Ui, _ctx: &mut MyContext) {
        let margin = Margin::same(4.0);

        Frame::none().inner_margin(margin).show(ui, |ui| {
            ui.heading("My egui Application");
            ui.horizontal(|ui| {
                ui.label("Your name: ");
                ui.text_edit_singleline(&mut self.title);
            });
            ui.add(Slider::new(&mut self.age, 0..=120).text("age"));
            if ui.button("Click each year").clicked() {
                self.age += 1;
            }
            ui.label(format!("Hello '{}', age {}", self.title, self.age));
        });
    }
}
```

So, let's look at this line:
```rs
 let margin = Margin::same(4.0);
 ```

Via this `Margin` we change inner padding to tab body.

**Comment by Jerrody** [comment]
So, it's just custom behavior, but we can make it much easier to do this.

**Comment by Adanos020** [comment]
Yup, I found that while working on my rewrite of `Tab` which I mentioned [here](https://github.com/Adanos020/egui_dock/issues/5) :+1:

**Comment by Jerrody** [comment]
> Yup, I found that while working on my rewrite of `Tab` which I mentioned [here](https://github.com/Adanos020/egui_dock/issues/5) 👍

Very cool!

**Comment by Adanos020** [comment]
Closed by #6
